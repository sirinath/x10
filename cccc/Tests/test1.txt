# statements in order, valid dependencies
0.0 w a 1
0.1 r b 0
1.0 w b 1
1.1 r a 0
2.0 r b 1

0.1 -> 1.0
1.1 -> 0.0
1.0 -> 2.0
