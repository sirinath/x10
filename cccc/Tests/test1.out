0.0 w a 1
0.1 r b 0
1.0 w b 1
1.1 r a 0
2.0 r b 1
# Manually specified dependencies
0.1 -> 1.0
1.0 -> 2.0
1.1 -> 0.0
# Program-order dependencies
0.0 -> 0.1
1.0 -> 1.1
# Propagated dependencies
0.0 -> 1.0
0.0 -> 1.1
0.1 -> 1.1
1.0 -> 0.0
1.0 -> 0.1
1.1 -> 0.1
Happens-before dependencies for task 0
0.0 -> 0.1
0.0 -> 1.0
0.0 -> 1.1
0.1 -> 1.0
0.1 -> 1.1
1.0 -> 0.0
1.0 -> 0.1
1.0 -> 1.1
1.1 -> 0.0
1.1 -> 0.1
Cycle located:
0.0 -> 0.1  (program order)
0.1 -> 1.0  (manual)
1.0 -> 0.0  (propagation)
Task 0 sees cycle
Happens-before dependencies for task 1
0.0 -> 0.1
0.0 -> 1.0
0.0 -> 1.1
0.1 -> 1.0
0.1 -> 1.1
1.0 -> 0.0
1.0 -> 0.1
1.0 -> 1.1
1.0 -> 2.0
1.1 -> 0.0
1.1 -> 0.1
Cycle located:
0.0 -> 0.1  (program order)
0.1 -> 1.0  (manual)
1.0 -> 0.0  (propagation)
Task 1 sees cycle
Happens-before dependencies for task 2
0.0 -> 0.1
1.0 -> 1.1
1.0 -> 2.0
Cycle found: illegal CCCC execution
